.github/workflows/lint-and-test.yml

```
@@ -23,17 +30,13 @@ jobs:
       - name: Install All Dependencies
         run: yarn
 
-      - name: Check Formatting with Prettier
-        run: yarn format-check-all
-
-      - name: Lint JS with ESLint
-        run: yarn lint
-
-      - name: Lint CSS with Stylelint
-        run: yarn lint-css
-
-      - name: Check TypeScript compilation
-        run: yarn lint-ts
+      - name: Lint and fix all changed files
+        run: yarn lint-staged -q --no-stash --diff "origin/main...HEAD"
```

```suggestion
        run: yarn lint-staged -q --no-stash --diff ${{ github.base_ref }}
```
I think it would be better to have this on other target branches as well not only on the `main` branch.